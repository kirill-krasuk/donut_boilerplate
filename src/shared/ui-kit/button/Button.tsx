import { ButtonSizes, ButtonStyles } from './enums';
import * as S                        from './styles';

import type { FC }                   from 'react';
import type { Props }                from './types';

/**
* This is Button component for donut boilerplate
*/
export const Button: FC<Props> = (props) => {
    const {
        onClick,
        children,
        size = ButtonSizes.Medium,
        style = ButtonStyles.Primary,
    } = props;

    return (
        <S.Wrapper
            cStyle={ style }
            size={ size }
            onClick={ onClick }
        >
            { children }
        </S.Wrapper>
    );
};

Button.defaultProps = {
    size : ButtonSizes.Medium,
    style: ButtonStyles.Primary
};

export default Button;
