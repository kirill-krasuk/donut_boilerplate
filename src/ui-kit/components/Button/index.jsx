// @flow

import React         from 'react';
import type { Node } from 'react';

import * as Styled   from './styled';

type PropsType = {

    /** custom onClick handler */

    onClick?: Function,

    /** possible button sizes */

    size: 'sm' | 'med' | 'lg',

    /** possible button colors themes */

    type: 'secondary' | 'primary',

    children: Node,
}

/**
 * basic UI elemnet of Donut Boilerplate
 * @visibleName Button
 * @version 1.0
 * @author [Kirill Krasuk](https://github.com/kirill-krasuk)
 */

const Button = (props: PropsType): Node => {
    const {
        onClick,
        size,
        children,
        type
    } = props;

    return (
        <Styled.Wrapper
            size={ size }
            onClick={ onClick }
            type={ type }
        >
            { children }
        </Styled.Wrapper>
    );
};

Button.defaultProps = {
    size: 'md',
    type: 'primary'
};

export default Button;
