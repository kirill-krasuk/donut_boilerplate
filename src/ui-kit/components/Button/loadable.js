// @flow

import React                  from 'react';
import type { ComponentType } from 'react';
import loadable               from '@loadable/component';

import type { PropsType }     from '.';

const component: ComponentType<PropsType> = loadable(
    () => import(/* webpackChunkName: "Button" */'.'),
    {
        fallback: <div>Loading..</div>
    }
);

export default component;
