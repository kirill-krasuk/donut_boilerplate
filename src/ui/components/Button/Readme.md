```jsx 
<Button size="sm" onClick={ () => console.log('heelllooooo') }>Hello</Button>
```

```jsx
<Button size="md" type="secondary">Hello</Button>
```

```jsx
<Button size="lg">Hello</Button>
```