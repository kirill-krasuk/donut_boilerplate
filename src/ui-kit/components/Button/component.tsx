import React, { FC }                 from 'react';
import { ButtonSizes, ButtonStyles } from '@ui-kit/enums/button';
import * as S                        from './styled';
import { Props }                     from './types';

/**
* This is Button component for donut boilerplate
*/
export const Button: FC<Props> = (props) => {
    const {
        onClick,
        children,
        size = ButtonSizes.Medium,
        style = ButtonStyles.Primary,
    } = props;

    return (
        <S.Wrapper
            size={ size }
            onClick={ onClick }
            cStyle={ style }
        >
            { children }
        </S.Wrapper>
    );
};

Button.defaultProps = {
    size : ButtonSizes.Medium,
    style: ButtonStyles.Primary
};

export default Button;
