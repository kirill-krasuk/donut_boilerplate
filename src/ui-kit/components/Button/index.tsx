import React             from 'react';

import { Sizes, Styles } from '@ui-kit/enums/button';
import * as S            from './styled';
import { Props }         from './types';

/**
 * basic UI element of Donut Boilerplate
 * @visibleName Button
 * @version 1.0
 * @author [Kirill Krasuk](https://github.com/kirill-krasuk)
 */

const Button: React.FC<Props> = (props): JSX.Element => {
    const {
        onClick,
        size,
        children,
        style,
    } = props;

    return (
        <S.Wrapper
            size={ size }
            onClick={ onClick }
            cStyle={ style }
        >
            { children }
        </S.Wrapper>
    );
};

Button.defaultProps = {
    size : Sizes.Medium,
    style: Styles.Primary
};

export default Button;
