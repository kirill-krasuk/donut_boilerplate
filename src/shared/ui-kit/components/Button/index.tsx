import loadable from '@loadable/component';

export default loadable(
    () => import(/* webpackChunkName: "Button" */'./component'),
    {
        fallback: <div>Loading..</div>
    }
);
