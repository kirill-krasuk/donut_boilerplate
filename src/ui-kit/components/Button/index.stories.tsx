import React            from 'react';

import { ButtonStyles } from '@ui-kit/enums/button';
import { Button }       from './component';

export default {
    title: 'Button',
};

export const withText = () => <Button style={ ButtonStyles.Primary }>Hello</Button>;

export const withEmoji = () => (
    <Button>
        <span role="img" aria-label="so cool">
            😀 😎 👍 💯
        </span>
    </Button>
);
