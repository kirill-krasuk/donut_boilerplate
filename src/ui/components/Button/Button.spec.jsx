import Button from '.';

it('should render a button', () => {
    const button = shallow(
        <Button>Click me</Button>
    );

    expect(button).toMatchSnapshot();
});
