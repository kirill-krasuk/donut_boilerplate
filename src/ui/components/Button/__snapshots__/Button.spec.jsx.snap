// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Button component Button init renders propery 1`] = `
<styled.button
  onClick={[Function]}
  size="sm"
  type="secondary"
>
  Test
</styled.button>
`;
